src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -489,12 +490,22 @@ public static SocketError WindowsIoctl(SafeSocketHandle handle, int ioControlCod
 
         public static unsafe SocketError SetSockOpt(SafeSocketHandle handle, SocketOptionLevel optionLevel, SocketOptionName optionName, int optionValue)
         {
-            SocketError errorCode = Interop.Winsock.setsockopt(
-                handle,
-                optionLevel,
-                optionName,
-                ref optionValue,
-                sizeof(int));
+            SocketError errorCode;
```

Comment:
I just noticed we're only doing this special logic in the SetSocketOption that takes an `int`.  What about the one that takes a `byte[]`? (Same question for GetSocketOption.)  While we'd certainly encourage using the `int`-based version, is there any reason we wouldn't expect it to behave similarly for the `byte[]` version?

Reply:
I didn't imagine people could choose the `byte[]` way, but you are right, nothing prevents this possibility